tee-worker/omni-executor/rpc-server/src/detailed_error.rs

```diff
@@ -0,0 +1,197 @@
+		Self::new(INVALID_ADDRESS_FORMAT_CODE, "Invalid address format")
+			.with_field(field)
+			.with_received(address.to_string())
+			.with_expected(expected_format)
+			.with_suggestion("Ensure the address follows the correct format")
+	}
+
+	pub fn invalid_amount(field: &str, amount: &str, reason: &str) -> Self {
+		Self::new(INVALID_AMOUNT_CODE, "Invalid amount value")
+			.with_field(field)
+			.with_received(amount.to_string())
+			.with_expected("Positive number within valid range")
+			.with_suggestion(reason)
+	}
+
+	pub fn invalid_hex_format(field: &str, value: &str, expected_length: Option<usize>) -> Self {
+		let expected = if let Some(len) = expected_length {
+			format!("0x-prefixed hex string of {} bytes", len)
+		} else {
+			"Valid hexadecimal string".to_string()
+		};
+
+		Self::new(INVALID_HEX_FORMAT_CODE, "Invalid hexadecimal format")
+			.with_field(field)
+			.with_received(value.to_string())
+			.with_expected(expected)
+			.with_suggestion("Check that the value is properly hex-encoded")
+	}
+
+	pub fn account_parse_error(account: &str, error: &str) -> Self {
+		Self::new(ACCOUNT_PARSE_ERROR_CODE, "Failed to parse account identifier")
+			.with_field("omni_account")
+			.with_received(account.to_string())
+			.with_expected("Valid 32-byte account identifier")
+			.with_suggestion(format!("Error: {}", error))
+	}
+
+	pub fn unexpected_response_type(expected: &str, received: &str) -> Self {
+		Self::new(UNEXPECTED_RESPONSE_TYPE_CODE, "Unexpected response type from service")
+			.with_expected(expected)
+			.with_received(received)
+			.with_suggestion("This is likely an internal error. Please contact support.")
+	}
+
+	pub fn signer_service_error(operation: &str, error: &str) -> Self {
+		Self::new(SIGNER_SERVICE_ERROR_CODE, "Signer service error")
+			.with_field("operation")
+			.with_received(operation.to_string())
+			.with_suggestion(format!("Signer error: {}", error))
+	}
+
+	pub fn email_service_error(_email: &str, _client_id: &str) -> Self {
+		Self::new(EMAIL_SERVICE_ERROR_CODE, "Email service error")
+			.with_field("email")
+			.with_suggestion("Failed to send verification email. Please try again later.")
+	}
+
+	pub fn storage_error(operation: &str, _key: &str) -> Self {
+		Self::new(STORAGE_SERVICE_ERROR_CODE, "Storage service error")
+			.with_field("operation")
+			.with_received(operation.to_string())
+			.with_suggestion("Storage operation failed. Please try again later.")
+	}
+
+	// Native task error factory methods
+	pub fn chain_not_supported(chain_id: u64) -> Self {
+		// Get supported chains from config
+		use crate::config::SUPPORTED_EVM_CHAINS;
+		let supported: Vec<u64> = SUPPORTED_EVM_CHAINS.iter().map(|&c| c as u64).collect();
+
+		Self::new(UNSUPPORTED_CHAIN_CODE, "Chain not supported")
+			.with_field("chain_id")
+			.with_received(chain_id.to_string())
+			.with_expected(format!("One of: {:?}", supported))
+			.with_suggestion("Please use a supported chain ID")
```

Comment:
There is already `INVALID_CHAIN_ID_CODE` , why we also need `UNSUPPORTED_CHAIN_CODE` ?